src/BaseModule.php

```diff
@@ -33,94 +34,80 @@
  *
  * @author Hypolite Petovan <hypolite@mrpetovan.com>
  */
-abstract class BaseModule
+abstract class BaseModule implements ICanHandleRequests
 {
+	/** @var array */
+	protected $parameters = [];
+
+	public function __construct(array $parameters = [])
+	{
+		$this->parameters = $parameters;
+	}
+
 	/**
-	 * Initialization method common to both content() and post()
-	 *
-	 * Extend this method if you need to do any shared processing before both
-	 * content() or post()
+	 * {@inheritDoc}
 	 */
-	public static function init(array $parameters = [])
+	public function init()
```

Comment:
The `init()` method isn't needed anymore since we have a constructor.

Reply:
I'd like to add constructor logic in a followup. We have to analyze if some of the `init()` if they are `rawContent()` because often they don't initialize the module, they execute business logic ;-)

Reply:
Not that I know of, `init()` have been used for processing common to `post()` and `content()` but `rawContent()` has always been on its own.

Reply:
https://github.com/friendica/friendica/blob/489cd0884ade34932fee45f136f77949bb3e3f91/src/Module/Magic.php#L42 for example :)